portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/pages/users/add.js

```diff
@@ -94,3 +120,13 @@ function userRegistration(claimMap, credentialMap, domain) {
         };
     }
 }
+
+function generatePassword(length) {
+    var length = length,
+        charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789",
+        retVal = "";
+    for (var i = 0, n = charset.length; i < length; ++i) {
+        retVal += charset.charAt(Math.floor(Math.random() * n));
```

Comment:
It's hard to achieve secure random from JS level. IMO, I don't think we need to secure this since this is directly going to the DB level and will not expose to the user/admin in any case. Need to discuss this in lead review.